lib/auth.guard.ts

```diff
@@ -60,11 +60,21 @@ function createAuthGuard(type?: string | string[]): Type<CanActivate> {
     }
 
     getRequest<T = any>(context: ExecutionContext): T {
-      return context.switchToHttp().getRequest();
+      if (context.getType() === 'http') {
+        return context.switchToHttp().getRequest();
+      } else if (context.getType<GqlContextType>() === 'graphql') {
+        const ctx = GqlExecutionContext.create(context);
+        return ctx.getContext().req;
+      }
```

Comment:
This should be left to each developer as they need. There's no reason to bring in the GraphQL package to the passport package